### DISCUSSION TITLE
Errors compiling the firmware code

### DISCUSSION THREAD

**Discussion by py2oc** [dl9rdz/rdz_ttgo_sonde#284] [TTGO firmware]
I've got some errors trying to compiling the code through Arduino IDE.
I'm newbie on C++ programming and need some help to understand why the errors.
Looks like a function is called multiple times with different parameters.

D:\Temp\rdz_ttgo_sonde-devel\RX_FSK\src\Display.cpp: In member function 'virtual void ILI9225Display::setFont(uint8_t)':
D:\Temp\rdz_ttgo_sonde-devel\RX_FSK\src\Display.cpp:534:27: error: call of overloaded 'setFont(NULL)' is ambiguous
  case 0: tft->setFont(NULL); tft->setTextSize(1); break;
                           ^
In file included from D:\Usuarios\Luiz\Documents\Arduino\libraries\GFX_Library_for_Arduino\src/Arduino_GFX_Library.h:28,
                 from D:\Temp\rdz_ttgo_sonde-devel\RX_FSK\src\Display.h:8,
                 from D:\Temp\rdz_ttgo_sonde-devel\RX_FSK\src\Display.cpp:7:
D:\Usuarios\Luiz\Documents\Arduino\libraries\GFX_Library_for_Arduino\src/Arduino_GFX.h:196:8: note: candidate: 'void Arduino_GFX::setFont(const GFXfont*)'
   void setFont(const GFXfont *f = NULL);
        ^~~~~~~
D:\Usuarios\Luiz\Documents\Arduino\libraries\GFX_Library_for_Arduino\src/Arduino_GFX.h:198:8: note: candidate: 'void Arduino_GFX::setFont(const uint8_t*)'
   void setFont(const uint8_t *font);
        ^~~~~~~

73 de PY2OC, Luiz

**Comment by dl9rdz** [comment]
You might go back to the previous version of the Arduino_GFX_Library. v1.1.8 should be ok

Or, I tried to make a small change that should fix this problem. In devel branch, but untested.

**Reply by dl9rdz** [reply]
The small change does not work on Linux, due to another problem with the new gfx library version (see https://github.com/moononournation/Arduino_GFX/issues/138). So I am not testing it right now. It might be ok on Windows or MacOS

But the downgrade to 1.1.8 should be fine.

**Comment by py2oc** [comment]
Hi Hansi.

Downgrading the Arduino_GFX library to v1.1.8 makes the compilation work.
Got few warnings but compiled successfully.
Thanks for help.

73 de PY2OC, Luiz
